include/nbl/builtin/glsl/ext/FFT/fft.glsl

```diff
@@ -0,0 +1,376 @@
+            for(uint t = 0u; t < num_virtual_threads; t++)
+            {
+                uint tid = thread_offset + t * _NBL_GLSL_EXT_FFT_WORKGROUP_SIZE_;
+                
+                uint even_index = nbl_glsl_ext_FFT_getEvenIndex(tid, 0, dataLength); // same as tid * 2
+                uint odd_index = even_index + 1;
+
+                _NBL_GLSL_SCRATCH_SHARED_DEFINED_[even_index] = floatBitsToUint(even_values[t][ch].x);
+                _NBL_GLSL_SCRATCH_SHARED_DEFINED_[odd_index] = floatBitsToUint(odd_values[t][ch].x);
+            }
+
+            barrier();
+            memoryBarrierShared();
+
+            for(uint t = 0u; t < num_virtual_threads; t++)
+            {
+                uint tid = thread_offset + t * _NBL_GLSL_EXT_FFT_WORKGROUP_SIZE_;
+
+                uint even_index = nbl_glsl_ext_FFT_getEvenIndex(tid, 0, dataLength); // same as tid * 2
+                uint odd_index = even_index + 1;
+
+                uint even_rev_bits = bitfieldReverse(even_index) >> leadingZeroes;
+                uint odd_rev_bits = bitfieldReverse(odd_index) >> leadingZeroes;
+
+                even_values[t][ch].x = uintBitsToFloat(_NBL_GLSL_SCRATCH_SHARED_DEFINED_[even_rev_bits]);
+                odd_values[t][ch].x = uintBitsToFloat(_NBL_GLSL_SCRATCH_SHARED_DEFINED_[odd_rev_bits]);
+            }
+
+            barrier();
+            memoryBarrierShared();
+            
+            // Get Even/Odd Values Y for virtual threads
+            for(uint t = 0u; t < num_virtual_threads; t++)
+            {
+                uint tid = thread_offset + t * _NBL_GLSL_EXT_FFT_WORKGROUP_SIZE_;
+                
+                uint even_index = nbl_glsl_ext_FFT_getEvenIndex(tid, 0, dataLength); // same as tid * 2
+                uint odd_index = even_index + 1;
+
+                _NBL_GLSL_SCRATCH_SHARED_DEFINED_[even_index] = floatBitsToUint(even_values[t][ch].y);
+                _NBL_GLSL_SCRATCH_SHARED_DEFINED_[odd_index] = floatBitsToUint(odd_values[t][ch].y);
+            }
+
+            barrier();
+            memoryBarrierShared();
+
+            for(uint t = 0u; t < num_virtual_threads; t++)
+            {
+                uint tid = thread_offset + t * _NBL_GLSL_EXT_FFT_WORKGROUP_SIZE_;
+
+                uint even_index = nbl_glsl_ext_FFT_getEvenIndex(tid, 0, dataLength); // same as tid * 2
+                uint odd_index = even_index + 1;
+
+                uint even_rev_bits = bitfieldReverse(even_index) >> leadingZeroes;
+                uint odd_rev_bits = bitfieldReverse(odd_index) >> leadingZeroes;
+
+                even_values[t][ch].y = uintBitsToFloat(_NBL_GLSL_SCRATCH_SHARED_DEFINED_[even_rev_bits]);
+                odd_values[t][ch].y = uintBitsToFloat(_NBL_GLSL_SCRATCH_SHARED_DEFINED_[odd_rev_bits]);
+            }
+
+            barrier();
+        }
+    }
+
+    // For loop for each stage of the FFT (each virtual thread computes 1 buttefly)
+	for(uint i = 0u; i < logTwo; ++i) 
```

Comment:
what I imagined it
```glsl
for (uint t=0u; t<num_virtual_threads; t++)
{
    nbl_glsl_ext_workgroup_FFT(even_values[t],odd_values[t]); // remember both parameters are `inout`
    barrier(); // because nbl_glsl_ext_workgroup_FFT reuses the same shared memory
}

for(uint i = findMSB(_NBL_GLSL_EXT_FFT_WORKGROUP_SIZE_-1u); i < logTwo; ++i)
{
    // do it without barriers and using shared memory
}
```